### DISCUSSION TITLE
How could I shuffle train index in every epoch with BatchSampler（DistributedSampler）

### DISCUSSION THREAD

**Discussion by xuanyang19** [Lightning-AI/pytorch-lightning#14462] [code help: CV]
How could I call set_epoch() on DistributedSampler in pytorch lightning? Although I set distributedsampler(shuffle = True), the train index is always the same at every epoch. What should I do to make the train index become different at every epoch?
my dataloader:
DataLoader(self.trainset,sampler=BatchSampler(DistributedSampler(self.trainset, shuffle=True), batch_size=self.batch_size,drop_last=False), num_workers=self.num_workers)

**Comment by rohitgr7** [comment]
it's not required. It will be done automatically for you within Lightning Trainer.
